### DISCUSSION TITLE
How to recognize disconnected shapes from booleanops

### DISCUSSION THREAD

**Discussion by Sieff** [ManimCommunity/manim#2857] [Q&A] [answered]
Hey, I'm currently trying to use manim to create animations for my upcoming seminar talk, where I talk about the packing anchored rectangles problem (https://doi.org/10.1007/s00493-015-3006-1). I'm completely new to manim and only used it for 3 days now, so I'm glad for any help.

What can you see in my render: The actual thing is the coordinate system with the square and dots. In the packing anchored rectangles problem, you can 'draw' rectangles from each dot towards the top right, while staying inside the outer square and not overlapping with any other dot/rectangle. I want to render the greedy approach for this problem, which is starting from the top-right corner towards the bottom left and drawing the largest possible rectangle for each dot.
Now we get into my problem: I want to look at the space, which is not yet inside a rectangle, for that I take the difference between the already occupied space and the rectangle, which spans from the current dot to the top-right corner. You can see this shape in white directly above the coordinate system.
My problem now is, that sometimes this shape is not a connected polygon. And I can't see how one can recognize from the Difference object, how the actual polygram looks like. I've looked a lot at the 'points' attribute from the Difference object, but I can't recognize when a new polygon will start. 
In the video f.e. the 8th point has this problem where the white shape above isn't connected and the blue shapes to the right, which are results from my calculations, now have a wrong shape.

https://user-images.githubusercontent.com/50653952/176965954-25c1efed-ea11-473a-b291-30bdac468cbb.mp4

TL;DR: How can you determine disconnected shapes from a booleanOps object?

**Comment by Sieff** [comment, marked as answer]
Ok, I found a solution for myself, sleeping over it really does help!
Yesterday I already read that the booleanOps are represented as Bézier curves. Didn't think hard enough about it...

Well today I looked again and saw that at least for my orthogonal shapes each line consists of 4 points, not sure if this holds true for more complex shapes. Well, so iterated along pairs of 4 points to get all the lines, then connected the lines into shapes by checking when a closed loop occurs. Afterwards, I just check which of the shapes contains my starting point to get the shape that I need.

Well then have a nice day everyone.
